client/src/providers/HealthProvider.tsx

```diff
@@ -131,47 +139,112 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
-      const signer = provider.getSigner();
-      const contract: HealthBlock = await fetchContract(signer);
+    const registerHCProviderHealthBlockContract = async (name: string, email: string, address: string, phone: string) => {
+        try {
+            const web3modal = new Web3Modal();
+            const connection = await web3modal.connect();
+            console.log(connection);
+            const provider = new ethers.providers.Web3Provider(connection);
+            console.log(provider);
+            const signer = provider.getSigner();
+            const contract: HealthBlock = await fetchContract(signer);
 
-      console.log(contract);
-      const register = await contract.registerHCProvider(name, email, address, phone);
-      register.wait();
-      console.log(register);
-    } catch (err) {
-      setError('Error Loading Health Contract');
+            console.log(contract);
+            const register = await contract.registerHCProvider(name, email, address, phone);
+            register.wait();
+            console.log(register);
+        }
+        catch (err) {
+            console.log(err)
+            setError("Error Loading Health Contract")
+        }
     }
-  };
 
-  return (
-    <HealthContext.Provider
-      value={{
-        checkIfWalletIsConnected,
-        connectWallet,
-        currentAccount,
-        healthBlockContract,
-        registerHealthBlockContract,
-        fetchPatientContract,
-        fetchPatientInfoContract,
-        registerDoctorHealthBlockContract,
-        registerHCProviderHealthBlockContract,
-      }}
-    >
-      {children}
-    </HealthContext.Provider>
-  );
+    const handleRaiseRequest = async (doctorName: string, file: File | null) => {
+        if (!file) return;
+        console.log("doctor name", doctorName)
+        const credentialsHash: string = "3423dg43";
+        console.log("credentialsHash", credentialsHash)
+        //const credentialsHash: Promise<string>  = getFileHash(file);
+        setDoctorName(doctorName);
+        const web3modal = new Web3Modal();
+        const connection = await web3modal.connect();
+        const provider = new ethers.providers.Web3Provider(connection);
+        const signer = provider.getSigner();
+        const contract = await fetchContract(signer);
+        
+        const tx = await contract.raiseRequest(doctorName, credentialsHash);
+        await tx.wait();
+        setDoctorName("");
+        setFile(null);
+        fetchRequests();
+    };
+
+    const handleApproveRequest = async (requestId: number) => {
+        try {
+          const web3modal = new Web3Modal();
+          const connection = await web3modal.connect();
+          const provider = new ethers.providers.Web3Provider(connection);
+          const signer = provider.getSigner();
+          const contract = await fetchContract(signer);
+          const tx = await contract.approveRequest(requestId);
+          await tx.wait();
+          fetchRequests();
```

Comment:
is this call necessary? can you explain why?

Reply:
yes, it is necessary to set verification request after each approval/rejection